uwblueprint/starter-code-v2
# Migrate Models To Prisma

## Notion ticket link

[Migrate all models into schema.prisma](https://www.notion.so/uwblueprintexecs/c5cb446b850a4b9a90e2044047142b41?v=9248f732f8cd4f409fb6e05500d9072d&p=b7180eb6a8684a0b8c6031a769898779&pm=s)



## Implementation description
* Defined the prisma models in schema.prisma
* Set up the database connection (env's in slack)
* Installed prisma dependencies
* Applied initial migration


## Steps to test
1. cd into the starter-code-cli repository and run `npm run dev -- --testing`
2. select (PERN and opt-in to all features) and clone from `cynthia/migrate-models-to-prisma`
4. add the .env files in the root, frontend/, and backend/typescript/models/primsa/
5. run `docker exec -it scv2_ts_backend bash` and cd into models/prisma
8. run `npx prisma migrate status` , if the initial migration has not been applied yet run `npx prisma migrate dev --name init`
9. run`npx prisma studio`
11. verify that the tables are loaded at http://localhost:5555/


## What should reviewers focus on?
* please also skim over the models and ensure they are the same as the sequelize models 
* when I introspected the database it automatically created the "created at" and "updated at" fields - do we want to keep these or not?

## Checklist
